lib/BrizyForms/Service/WebHooksService.php

```diff
@@ -0,0 +1,238 @@
+<?php
+
+namespace BrizyForms\Service;
+
+use BrizyForms\Exception\ServiceException;
+use BrizyForms\FieldMap;
+use BrizyForms\Model\Account;
+use BrizyForms\Model\Folder;
+use BrizyForms\Model\Group;
+use BrizyForms\Model\GroupData;
+use BrizyForms\Model\RedirectResponse;
+use BrizyForms\Model\Response;
+use BrizyForms\ServiceConstant;
+use GuzzleHttp\Client;
+class WebHooksService extends Service
+{
+
+    private static function curlExec($value)
+    {
+        $client = new Client();
+
+        $options = array();
+        if(array_key_exists('header', $value))
+        {
+            $options['headers'] = $value['header'] ;
+        }
+
+        if(array_key_exists('message', $value)) {
+            $options['form_params'] = $value['message'];
+        }
+
+        $client->request($value['sendType'], $value['url'], $options);
+
+        return true;
+    }
+
+    /**
+     * @param FieldMap $fieldMap
+     * @param string $group_id
+     *
+     * @return mixed
+     */
+    protected function mapFields(FieldMap $fieldMap, $group_id = null)
+    {
+        foreach ($fieldMap->toArray() as $fieldLink) {
+            if ($fieldLink->getTarget() == ServiceConstant::AUTO_GENERATE_FIELD) {
+                $fieldLink->setTarget($fieldLink->getSourceTitle());
+            }
+        }
+        return $fieldMap;
+    }
+
+    /**
+     * @param FieldMap $fieldMap
+     * @param null $group_id
+     * @param array $data
+     * @param bool $confirmation_email
+     * @return mixed|void
+     * @throws ServiceException
+     * @throws \BrizyForms\Exception\FieldMapException
+     */
+    protected function internalCreateMember(FieldMap $fieldMap, $group_id = null, array $data = [], $confirmation_email = false)
+    {
+        $data = $fieldMap->transform($data, false);
+
+        $email = [];
+        if ($data->getEmail()) {
+            $email = ['Email' => $data->getEmail()];
+        }
+
+        $data_json = json_encode(array_merge($email, $data->getFields()));
+        $auth_data = $this->authenticationData->getData();
+
+        $sends = ['url' => $auth_data['urlHooks'], 'sendType' => $auth_data['sendType'], 'message' => $email];
+        if (array_key_exists( 'header', $auth_data))
+        {
+            $sends['header'] = $auth_data['header'];
+        }
+
+        if (!self::curlExec($sends)) {
+            throw new ServiceException('Can\'t send data to this webhook_url');
+        }
+    }
+
+    protected function curlII($url)
+    {
+        $ch = curl_init($url);
+
```

Comment:
or use the curl for both cases (auth && send request) or use the GuzzleHttp\Client